### ISSUE TITLE
Downgrade the dependency for public_suffix

### ISSUE THREAD

**Issue by rumankazi** [sporkmonger/addressable#541]
https://github.com/sporkmonger/addressable/blob/7930ece6e4ae266e67fc927f2e44e39f97fd5f22/addressable.gemspec#L26

Please downgrade this to `< 6.0`

Since `public_suffix >= 6.0` requires `ruby >= 3.0`,
the latest release of addressable is breaking for `ruby < 3.0`.
@dentarg 

Error:
```
 The last version of public_suffix (>= 2.0.2, < 7.0) to support your Ruby & RubyGems was 5.1.1. Try installing it with `gem install public_suffix -v 5.1.1` and then running the current command again
        public_suffix requires Ruby version >= 3.0. The current ruby version is 2.7.5.203
```

**Comment by dentarg** [comment]
How are you installing addressable?

**Comment by dentarg** [comment]
We have been here before: https://github.com/sporkmonger/addressable/issues/476

**Comment by dentarg** [comment]
Update RubyGems with `gem update --system 3.4.22` and it will work


```
$ docker run --rm -it -v $(pwd):/app -w /app ruby:2.7.8 bash
root@9c2ecd0652c4:/app# gem -v
3.1.6

root@9c2ecd0652c4:/app# gem install addressable
Fetching public_suffix-6.0.0.gem
Fetching addressable-2.8.7.gem
ERROR:  Error installing addressable:
       The last version of public_suffix (>= 2.0.2, < 7.0) to support your Ruby & RubyGems was 5.1.1. Try installing it with `gem install public_suffix -v 5.1.1` and then running the current command again
       public_suffix requires Ruby version >= 3.0. The current ruby version is 2.7.8.225.

root@9c2ecd0652c4:/app# gem update --system 3.4.22
Updating rubygems-update
Fetching rubygems-update-3.4.22.gem
Successfully installed rubygems-update-3.4.22
Installing RubyGems 3.4.22
  Successfully built RubyGem
  Name: bundler
  Version: 2.4.22
  File: bundler-2.4.22.gem
Bundler 2.4.22 installed
RubyGems 3.4.22 installed
Regenerating binstubs
Regenerating plugins

...


------------------------------------------------------------------------------

RubyGems installed the following executables:
       /usr/local/bin/gem
       /usr/local/bin/bundle
       /usr/local/bin/bundler

RubyGems system software updated

root@9c2ecd0652c4:/app# gem -v
3.4.22
root@9c2ecd0652c4:/app# gem install addressable
Fetching public_suffix-5.1.1.gem
Successfully installed public_suffix-5.1.1
Successfully installed addressable-2.8.7
2 gems installed

A new release of RubyGems is available: 3.4.22 → 3.5.14!
Run `gem update --system 3.5.14` to update your installation.
```
